kryo-serializer/src/test/java/org/vertexium/serializer/kryo/KryoVertexiumSerializerTest.java

```diff
@@ -218,23 +218,24 @@ public void before() {
             111, 32, 99, 111, 108, 108, 101, 99, 116, 105, 111, 110, 32, 119, 105, 116,
             104, 32, 100, 101, 115, 99, 114, 105, 112, 116, 105, 111, -18, 1, 1, 106,
             97, 118, 97, 46, 117, 116, 105, 108, 46, 65, 114, 114, 97, 121, 76, 105,
-            115, -12, 1, 2, 1, 1, 1, 4, -21, 7, 1, 0, -128, -47, 34, -37,
-            -7, 126, -78, 83, 64, 41, 92, -113, -62, -11, 120, 86, 64, -21, 7, 1,
-            0, -128, 96, -27, -48, 34, -37, 105, 80, 64, 47, -35, 36, 6, -127, 69,
-            75, 64, -21, 7, 1, 0, -128, -98, -17, -89, -58, 75, -73, 69, 64, 12,
-            2, 43, -121, 22, 41, 64, 64, -21, 7, 1, 0, -128, -47, 34, -37, -7,
-            126, -78, 83, 64, 41, 92, -113, -62, -11, 120, 86, 64, 1, 1, 1, 4,
-            -21, 7, 1, 0, -128, -10, 40, 92, -113, -62, 53, 53, 64, 43, -121, 22,
-            -39, -50, 55, 36, 64, -21, 7, 1, 0, -128, -14, -46, 77, 98, 16, -8,
-            85, 64, 41, 92, -113, -62, -11, 48, 83, 64, -21, 7, 1, 0, -128, 96,
-            -27, -48, 34, -37, 105, 80, 64, 47, -35, 36, 6, -127, 69, 75, 64, -21,
-            7, 1, 0, -128, -10, 40, 92, -113, -62, 53, 53, 64, 43, -121, 22, -39,
-            -50, 55, 36, 64, 1, 1, 1, 4, -21, 7, 1, 0, -128, -27, -48, 34,
-            -37, -7, 62, 40, 64, -106, 67, -117, 108, -25, 59, 55, 64, -21, 7, 1,
-            0, -128, 92, -113, -62, -11, 40, 44, 65, 64, -18, 124, 63, 53, 94, -70,
-            70, 64, -21, 7, 1, 0, -128, 127, 106, -68, 116, -109, 72, 76, 64, 8,
-            -84, 28, 90, 100, -21, 80, 64, -21, 7, 1, 0, -128, -27, -48, 34, -37,
-            -7, 62, 40, 64, -106, 67, -117, 108, -25, 59, 55, 64, 69, 78, -60
+            115, -12, 1, 2, 1, 1, 1, 4, -9, 7, 1, 0, 0, -128, 125, 63,
```

Comment:
You shouldn't change these bytes here. Create a new set of bytes if you want but these bytes represent what could be in the database and to verify that they will deserialize properly.

Reply:
I think I have to change these bytes. [The polygon being used in the base class is actually invalid](https://github.com/visallo/vertexium/blob/master/test/src/main/java/org/vertexium/test/VertexiumSerializerTestBase.java#L229-L251). The coordinates of the holes are going in the wrong direction and the holes aren't within the outer shell.

Reply:
We should handle that case though since all the existing data is going to have it like that. My intent for these byte arrays is that they represent what could exist in real databases.

Reply:
I was worried about that too. The existing data deserializes just fine with these changes. The [issue is here where the test tries to build a polygon](https://github.com/visallo/vertexium/blob/master/test/src/main/java/org/vertexium/test/VertexiumSerializerTestBase.java#L229-L251) to compare the bytes to. In order to verify, I fixed that polygon but left the bytes the same. Deserialization worked and [the test failed here](https://github.com/visallo/vertexium/blob/master/test/src/main/java/org/vertexium/test/VertexiumSerializerTestBase.java#L261) trying to compare the coordinates.

Reply:
Fixed. I have new legacy methods that test deserializing the exact same bytes to ensure it still works. In addition, I added a new method in GraphTestBase that tests mutating and re-indexing a legacy vertex with invalid geometry.